lib/mail/matchers/has_sent_mail.rb

```diff
@@ -51,6 +51,10 @@ def bcc(recipient_or_list)
         self
       end
 
+      def with_attachments(*attachments)
+        @attachments = attachments
+        self
+      end
```

Comment:
There's more than one route to implement this in Ruby. You can use splats so long as this method behaves like the rest of the API.

Is the idiomatic way to pass an "any" matcher rather than treating empty splat args as an implied "any," e.g.
```ruby
is_expected.to have_sent_email.with_attachments(any_attachment)
```